ext/semian/sysv_semaphores.c

```diff
@@ -80,19 +81,27 @@ set_semaphore_permissions(int sem_id, long permissions)
 }
 
 int
-perform_semop(int sem_id, short index, short op, short flags, struct timespec *ts)
+perform_semop(int sem_id, short index, short op, short flags, struct timespec *ts, int *wait_time_ms)
 {
   struct sembuf buf = { 0 };
 
   buf.sem_num = index;
   buf.sem_op  = op;
   buf.sem_flg = flags;
 
+  struct timespec begin, end;
+  if (wait_time_ms) {
```

Comment:
I'd suggest moving both this and the timing finish to a helper function, as this complicates the control flow for a bit